transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,155 @@
+= Support Transactions during WildFly’s graceful shutdown
+:author:        	Manuel Finelli
+:email:         	jfinelli@redhat.com
+:toc:           	left
+:icons:         	font
+:idprefix:
+:idseparator:   	-
+
+== Overview
+
+When employing WildFly in cloud environments, distributed transactions are not supported out of the box. In particular, Narayana (and, thus, the Transactions subsystem) has some [hard requirements](https://jbossts.blogspot.com/2022/04/narayana-on-cloud-part-1.html) that need to be satisfied. The goal of this proposal is to outline how distributed transactions should be safely terminated during the graceful shutdown of WildFly, with a particular focus on cloud environments.
+
+=== Current Issues
+
+_1.1 Currently, WildFly’s graceful shutdown does not take into consideration that hosting entities can be shut down indefinitely, i.e. there will not be any restart/resume after shutdown/suspension. From the point of view of the Transactions subsystem, this scenario might develop into a data integrity issue. In fact, Narayana’s Recovery Coordinator does not necessarily resolve all in-doubt transactions during its suspension, leaving the resolution of left-over transactions for later, when the Recovery Coordinator gets resumed._
```

Comment:
Not completing in doubt transactions does not impact data integrity, only availability. The proposal mentions  data integrity in a number of places - it should use different terminology.

Reply:
I'm not clear on what your response means but, since a heuristic outcome cannot be resolved, it's only relevance to graceful shutdown is that the details of the resource that reported it are captured somewhere, either in log output that gets backed up or in an object store. But I would keep any reference to heuristics in a separate section and not scatter it throughout the document or labour the point - this stuff is hard enough to parse without needing to infer its meaning.

Reply:
I'll think about your comment and I'll try to address it ASAP. Thanks for your feedback